bundles/core/org.eclipse.smarthome.core.persistence/src/main/java/org/eclipse/smarthome/core/persistence/rest/PersistenceResource.java

```diff
@@ -0,0 +1,261 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.persistence.rest;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.persistence.FilterCriteria;
+import org.eclipse.smarthome.core.persistence.FilterCriteria.Ordering;
+import org.eclipse.smarthome.core.persistence.HistoricItem;
+import org.eclipse.smarthome.core.persistence.PersistenceService;
+import org.eclipse.smarthome.core.persistence.QueryablePersistenceService;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.io.rest.RESTResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+/**
+ * This class acts as a REST resource for history data and provides different methods to interact with the persistence
+ * store
+ *
+ * @author Chris Jackson - Initial Contribution
+ */
+@Path(PersistenceResource.PATH)
+@Api(value = PersistenceResource.PATH)
+public class PersistenceResource implements RESTResource {
+
+    private final Logger logger = LoggerFactory.getLogger(PersistenceResource.class);
+
+    // The URI path to this resource
+    public static final String PATH = "persistence";
+
+    static private Map<String, PersistenceService> persistenceServices = new HashMap<String, PersistenceService>();
+
+    public void addPersistenceService(PersistenceService service) {
+        persistenceServices.put(service.getName(), service);
+    }
+
+    public void removePersistenceService(PersistenceService service) {
+        persistenceServices.remove(service.getName());
+    }
+
+    @GET
+    @Produces({ MediaType.APPLICATION_JSON })
+    @ApiOperation(value = "Gets a list of persistence services.", response = String.class, responseContainer = "List")
+    @ApiResponses(value = @ApiResponse(code = 200, message = "OK") )
+    public Response httpGetPersistenceServices(@Context HttpHeaders headers) {
+        Object responseObject = getPersistenceServiceList();
+        return Response.ok(responseObject).build();
+    }
+
+    @GET
+    @Path("/{servicename: [a-zA-Z_0-9]*}/{itemname: [a-zA-Z_0-9]*}")
```

Comment:
In order to retrieve the configured defaultService, we should probably move the configuration from PersistenceExtensions.defaultService to some new exported class in core.persistence, so that this info can be used from different places.